robotiq_hande_driver/src/protocol_logic.cpp

```diff
@@ -0,0 +1,134 @@
+#include "protocol_logic.hpp"
+
+#include <cstdio>
+
+
+namespace hande_driver  {
+
+constexpr auto kGripperPositionOpenedThreshold = 230;
+constexpr auto kGripperPositionClosedThreshold = 13;
+
+ProtocolLogic::ProtocolLogic()
+:   status_(0)
+,   activation_status_(GRIPPER_RESET)
+,   action_status_(STOPPED)
+,   gripper_status_(NOT_USED)
+,   object_detection_status_(REQ_POS_NO_OBJECT)
+,   fault_status_(0)
+,   position_request_echo_(0)
+,   position_(0)
+,   current_(0)
```

Comment:
* We can omit zeros for initialization ([for the zero value](https://en.cppreference.com/w/cpp/language/zero_initialization)) .
* Thanks to the `enum class` it's now clear which enum is being used.
```suggestion
:   status_()
,   activation_status_(ActivationStatus::GRIPPER_RESET)
,   action_status_(ActionStatus::STOPPED)
,   gripper_status_(GripperStatus::NOT_USED)
,   object_detection_status_(ObjectDetectionStatus::REQ_POS_NO_OBJECT)
,   fault_status_()
,   position_request_echo_()
,   position_()
,   current_()
```